crates/hc_sandbox/src/calls.rs

```diff
@@ -440,22 +430,24 @@ async fn call_inner(cmd: &mut CmdRunner, call: AdminRequestCli) -> anyhow::Resul
             );
         }
         AdminRequestCli::DumpNetworkStats => {
-            let stats = dump_network_stats(cmd).await?;
+            let stats = client.dump_network_stats().await?;
             // Print without other text so it can be piped
             println!("{}", serde_json::to_string(&stats)?);
         }
         AdminRequestCli::ListCapabilityGrants(args) => {
-            let info = list_capability_grants(cmd, args).await?;
+            let info = client
+                .list_capability_grants(args.installed_app_id, args.include_revoked)
+                .await?;
             // Print without other text so it can be piped
             println!("{:?}", info);
         }
-        AdminRequestCli::AddAgents => todo!("Adding agent info via CLI is coming soon"),
+        AdminRequestCli::AddAgents => todo!("Adding agent info via CLI is not implemented"),
```

Comment:
Any reason not to deal with this TODO? Probably not a priority because it's not commonly used but it would be good if it was possible through the Rust client at least?

Reply:
Yes, why not. It is possible via the Rust client already...I guess we could just make this a string parameter that takes an encoded `AgentInfoSigned` (or multiple comma separated ones)?

Reply:
I think it's easiest if it matches the agent info output of the conductor but now I'm realizing that is probably missing here too. That should be a JSON array of objects, as a string

Reply:
Would that need to be readable from a file then? Or do we expect this to be passed as a CLI argument directly? `AgentInfoSigned` is quite a large object after all...

Reply:
It'd be nice to support stdin so that people can do a bit of shell work to get the value out of a file but I think CLI is acceptable as a starting point. It's what I ended up doing for zome call JSON payloads -> https://github.com/holochain/holochain/blob/develop/crates/hc_sandbox/src/zome_call.rs#L149

Reply:
Oh I remember why, we had to accept a password already on stdin which isn't a problem here. So yes, maybe use stdin? Then it's flexible about whether people want to actually use a file or not

Reply:
> we had to accept a password already on stdin which isn't a problem here

I think that's exactly the same issue here unfortunately. The password is still required for `hc-sandbox call` and it can be provided [here](https://github.com/holochain/holochain/blob/d66e0611743c41bbea0a09da0f625eae35254bc4/crates/hc_sandbox/src/cli.rs#L29) at the root level hc-sanbox command.

Reply:
It's present as a global argument but does it always get read/prompted for even when it's not needed? I think that should only be used for `hc s run`, `hc s create` and `hc s generate`?

Reply:
It's being used with `hc s call` as well in case there's no conductor running yet because in that case, a conductor gets started [here](https://github.com/holochain/holochain/blob/d986d0c8f4ac73b314c6eb534eb756755dd36628/crates/hc_sandbox/src/calls.rs#L307) (after reading the `.hc` file in the current working directory to retrieve the conductor environment path) which then reads the password from stdin [here](https://github.com/holochain/holochain/blob/d986d0c8f4ac73b314c6eb534eb756755dd36628/crates/hc_sandbox/src/run.rs#L98).

Reply:
I also added a test for this scenario now 141ab6a91cb1511e3e5c785c66571f3d8038076b which succeeds on develop with the previous `CmdRunner` logic but currently fails with the changes of this PR with `Websocket error: Websocket closed: ConnectionClosed` which I don't understand yet.

Reply:
Okay, added it here: a924259afb45023ba6e3a3147e7b5cd489f99fb2